themes/default/content/blog/changes-to-import/index.md

```diff
@@ -0,0 +1,373 @@
+
+### Terraform `Read`
+
+Many of Pulumi's resource providers make use of Terraform providers. We call these bridged providers and all of them use our "terraform-bridge" library to translate between the Terraform API and Pulumi API.
+
+Terraform currently has a slightly odd behavior that missing values for some properties are zero initialized by its `Refresh` method (what we use to implement Pulumi's `Read` method in the terraform bridge). This has the result that when Pulumi asks to read a given Terraform resource some top level properties (such as S3 buckets `acl` property) come back as missing, while other top level properties and all nested properties (such as S3 buckets `versioning.mfaDelete`) come back with some zero value.
+
+For example reading an AWS S3 bucket object with the current Terraform bridge results in the following property set:
+
+```
+{
+    "__defaults": [],
+    "accelerationStatus": "",
+    "arn": "arn:aws:s3:::my-bucket-3f85c54",
+    "bucket": "my-bucket-3f85c54",
+    "corsRules": [],
+    "grants": [],
+    "hostedZoneId": "Z3BJ6K6RIION7M",
+    "lifecycleRules": [],
+    "loggings": [],
+    "objectLockConfiguration": null,
+    "replicationConfiguration": null,
+    "requestPayer": "BucketOwner",
+    "serverSideEncryptionConfiguration": null,
+    "tags": {
+        "__defaults": []
+    },
+    "versioning": {
+        "__defaults": [],
+        "enabled": false,
+        "mfaDelete": false
+    },
+    "website": null
+}
+```
+
+We've made [a change](https://github.com/pulumi/pulumi-terraform-bridge/pull/453) to the Terraform bridge to try and elide these zero properties that _should_ just be missing. Reading the same S3 bucket with these changes gives the following much smaller property set:
+
+```
+{
+    "__defaults": [],
+    "arn": "arn:aws:s3:::my-bucket-3f85c54",
+    "bucket": "my-bucket-3f85c54",
+    "hostedZoneId": "Z3BJ6K6RIION7M",
+    "requestPayer": "BucketOwner",
+    "website": null
+}
+```
+
+We'll continue to work on the Terraform bridge to give better results here. For example the `website` property on the bucket should probably be missing rather than null.
+
+## Feedback
+
+We're aware this could have a large impact to some of our users so we wanted to make sure we explained these changes well. If you have questions or worries about this change let us know at our community [Slack](https://slack.pulumi.com/) or [GitHub Discussions](https://github.com/pulumi/pulumi/discussions).
```

Comment:
```suggestion
We're aware this could have a large impact to some of our users so we wanted to make sure we explained these changes well. If you have questions or concerns about this change let us know in our community [Slack](https://slack.pulumi.com/) or [in GitHub Discussions](https://github.com/pulumi/pulumi/discussions).
```